src/NetTopologySuite/Triangulate/QuadEdge/Vertex.cs

```diff
@@ -30,9 +30,41 @@ public class Vertex : IEquatable<Vertex>
         private const int ORIGIN = 5;
         private const int DESTINATION = 6;
 
+
+        
+        private int id; // ID of the vertex	
         private readonly Coordinate _p;
+        private bool border;
         // private int edgeNumber = -1;
 
+        /// <summary>	
+        /// Default constructor	
+        /// </summary>	
+        public Vertex()
+        { }
+
+        /// <summary>	
+        /// Creates an instance of this class using the given id	
+        /// </summary>	
+        /// <param name="id">ID of the vertex</param>	
+        public Vertex(int id)
+        {
+            this.id = id;
+        }
+
+
+        /// <summary>	
+        /// Creates an instance of this class using the given id	
+        /// </summary>	
+        /// <param name="id">ID of the vertex</param>	
+        /// <param name="border">defines if the vertex is a border vertex or not in the triangulation framework</param>	
+        public Vertex(int id, bool border)
+        {
+            this.id = id;
+            this.border = border;
+        }
```

Comment:
"defines if the vertex is a border vertex or not"
we should have two classes: BorderVertex and Vertex . since we gonna have multiple types of vertices thus addition of an abstract class as a base for  BorderVertex and Vertex would be better.